rasa/core/events/__init__.py

```diff
@@ -98,38 +110,42 @@ def from_story_string(
         parameters: Dict[Text, Any],
         default: Optional[Type["Event"]] = None,
     ) -> Optional[List["Event"]]:
-        event = Event.resolve_by_type(event_name, default)
+        event_class = Event.resolve_by_type(event_name, default)
 
-        if event:
-            return event._from_story_string(parameters)
-        else:
+        if not event_class:
             return None
 
+        return event_class._from_story_string(parameters)
+
     @staticmethod
     def from_parameters(
         parameters: Dict[Text, Any], default: Optional[Type["Event"]] = None
     ) -> Optional["Event"]:
 
         event_name = parameters.get("event")
-        if event_name is not None:
-            copied = parameters.copy()
-            del copied["event"]
+        if event_name is None:
+            return None
 
-            event = Event.resolve_by_type(event_name, default)
-            if event:
-                return event._from_parameters(parameters)
-            else:
-                return None
-        else:
+        copied = parameters.copy()
+        del copied["event"]
+
+        event_class = Event.resolve_by_type(event_name, default)
+        if not event_class:
             return None
 
+        return event_class._from_parameters(parameters)
+
     @classmethod
     def _from_story_string(cls, parameters: Dict[Text, Any]) -> Optional[List["Event"]]:
         """Called to convert a parsed story line into an event."""
-        return [cls(parameters.get("timestamp"))]
+        return [cls(parameters.get("timestamp"), parameters.get("metadata"))]
 
     def as_dict(self):
-        return {"event": self.type_name, "timestamp": self.timestamp}
+        return {
+            "event": self.type_name,
+            "timestamp": self.timestamp,
+            "metadata": self.metadata,
```

Comment:
should we even return the `metadata` field in case it's empty? Would leave a smaller footprint (instead of having to store an empty dict for every event)

Reply:
We can do that, but with two exceptions: `BotUttered` and `UserUttered` already returned their `metadata` field unconditionally when serialized. I can keep this behaviour for these two events and make it so that it's only returned when present for all other events.

Reply:
Added a commit that implements this.